pkg/injection/namespace/bootstrapperconfig/bootstrapperconfig.go

```diff
@@ -0,0 +1,325 @@
+
+		return nil, err
+	}
+
+	pmc = pmc.
+		AddHostGroup(dk.OneAgent().GetHostGroup()).
+		AddConnectionInfo(dk.Status.OneAgent.ConnectionInfoStatus, tenantToken).
+		// set proxy explicitly empty, so old proxy settings get deleted where necessary
+		AddProxy("")
+
+	if dk.NeedsOneAgentProxy() {
+		proxy, err := dk.Proxy(ctx, g.apiReader)
+		if err != nil {
+			conditions.SetKubeApiError(dk.Conditions(), processmoduleconfigsecret.PmcConditionType, err)
+
+			return nil, err
+		}
+
+		pmc.AddProxy(proxy)
+
+		multiCap := capability.NewMultiCapability(&dk)
+		dnsEntry := capability.BuildDNSEntryPointWithoutEnvVars(dk.Name, dk.Namespace, multiCap)
+
+		if dk.FeatureNoProxy() != "" {
+			dnsEntry += "," + dk.FeatureNoProxy()
+		}
+
+		pmc.AddNoProxy(dnsEntry)
+	}
+
+	marshaled, err := json.Marshal(pmc)
+	if err != nil {
+		log.Info("could not marshal process module config")
+
+		return nil, err
+	}
+
+	return marshaled, err
+}
+
+func (g *BootstrapperInitGenerator) createSecretConfigForDynaKube(ctx context.Context, dk *dynakube.DynaKube, hostMonitoringNodes map[string]string) (*startup.SecretConfig, error) {
+	var tokens corev1.Secret
+	if err := g.client.Get(ctx, client.ObjectKey{Name: dk.Tokens(), Namespace: g.namespace}, &tokens); err != nil {
+		return nil, errors.WithMessage(err, "failed to query tokens")
+	}
+
+	var proxy string
+
+	var err error
+	if dk.NeedsOneAgentProxy() {
+		proxy, err = dk.Proxy(ctx, g.apiReader)
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+	}
+
+	oneAgentNoProxyValues := []string{}
+
+	if dk.NeedsCustomNoProxy() {
+		oneAgentNoProxyValues = append(oneAgentNoProxyValues, dk.FeatureNoProxy())
+	}
+
+	if dk.ActiveGate().IsRoutingEnabled() {
+		multiCap := capability.NewMultiCapability(dk)
+		oneAgentDNSEntry := capability.BuildDNSEntryPointWithoutEnvVars(dk.Name, dk.Namespace, multiCap)
+		oneAgentNoProxyValues = append(oneAgentNoProxyValues, oneAgentDNSEntry)
+	}
+
+	return &startup.SecretConfig{
+		ApiUrl:              dk.Spec.APIURL,
+		ApiToken:            getAPIToken(tokens),
+		PaasToken:           getPaasToken(tokens),
+		TenantUUID:          dk.Status.OneAgent.ConnectionInfoStatus.TenantUUID,
+		Proxy:               proxy,
+		NoProxy:             dk.FeatureNoProxy(),
+		OneAgentNoProxy:     strings.Join(oneAgentNoProxyValues, ","),
+		NetworkZone:         dk.Spec.NetworkZone,
+		SkipCertCheck:       dk.Spec.SkipCertCheck,
+		HasHost:             dk.OneAgent().IsCloudNativeFullstackMode(),
+		MonitoringNodes:     hostMonitoringNodes,
+		HostGroup:           dk.OneAgent().GetHostGroup(),
+		InitialConnectRetry: dk.FeatureAgentInitialConnectRetry(),
+		EnforcementMode:     dk.FeatureEnforcementMode(),
+		ReadOnlyCSIDriver:   dk.FeatureReadOnlyCsiVolume(),
+		CSIMode:             dk.OneAgent().IsCSIAvailable(),
+	}, nil
+}
+
+func getPaasToken(tokens corev1.Secret) string {
+	if len(tokens.Data[dtclient.PaasToken]) != 0 {
```

Comment:
this can panic, of Data doesn't have a required key.

Reply:
The function got removed anyway, as it is not needed for this secret https://github.com/Dynatrace/dynatrace-operator/pull/4515/commits/349bd42a7afbecd1850b936cfa50bc74e8f49892